apis/v1alpha1/types.go

```diff
@@ -290,6 +401,28 @@ type ThrottleSettings struct {
 	RateLimit  *float64 `json:"rateLimit,omitempty"`
 }
 
+// Represents a usage plan used to specify who can assess associated API stages.
+// Optionally, target request rate and quota limits can be set. In some cases
+// clients can exceed the targets that you set. Don’t rely on usage plans
+// to control costs. Consider using Amazon Web Services Budgets (https://docs.aws.amazon.com/cost-management/latest/userguide/budgets-managing-costs.html)
+// to monitor costs and WAF (https://docs.aws.amazon.com/waf/latest/developerguide/waf-chapter.html)
+// to manage API requests.
+type UsagePlan struct {
+	Description *string            `json:"description,omitempty"`
+	ID          *string            `json:"id,omitempty"`
+	Name        *string            `json:"name,omitempty"`
+	ProductCode *string            `json:"productCode,omitempty"`
+	Tags        map[string]*string `json:"tags,omitempty"`
+}
+
+// Represents a usage plan key to identify a plan customer.
+type UsagePlanKey struct {
+	ID    *string `json:"id,omitempty"`
+	Name  *string `json:"name,omitempty"`
+	Type  *string `json:"type_,omitempty"`
```

Comment:
i this is used in the crd/spec, we need to rename the tag:
```suggestion
	Type  *string `json:"type,omitempty"`
```

Reply:
Its not used in the crd